### PR TITLE
Named parameters for loading the images

### PR DESCRIPTION
**Pull request by astridx**
Setting Up Your Local Environment.

Unfortunately, the setting up of the local environment in Joomla 4 is not yet possible. Is that due to Composer 3 (https://github.com/joomla/joomla-cms/discussions/31783)?

But I was able to use Joomla 3 with PHP8.

Regarding the environmental variables: https://xdebug.org/docs/upgrade_guide#changed-xdebug.profiler_enable_trigger

What do you think?

### CODE REVIEW

**Comment by laoneo** [comment]
Nice.
Instead of making just one image with a named parameter, I would make different images, like that we can then switch for example in the webserver task easily from PHP 7 to 8. Otherwise you have to rebuild the image all the time when you want to switch.

File: images/cli/Dockerfile [hunk 1]
```diff
@@ -10,6 +14,8 @@ ENV PHP_EXTENSION_PDO_PGSQL=1
 
 # PHP ini variables
 ENV PHP_INI_XDEBUG__PROFILER_ENABLE_TRIGGER=1
+ENV PHP_INI_XDEBUG__MODE=profil
```
Comment (laoneo) [inline images/cli/Dockerfile:17]: Can you make a different PR for theses variables, so we can merge it already.

Comment (astridx) [reply images/cli/Dockerfile:17]: Please see https://github.com/Digital-Peak/DPDocker/pull/22

**Comment by laoneo** [comment]
Based on your pr here I made another one which creates an image per PHP version. Like that we can switch PHP versions easier without the need to rebuild the image. I made it based on your idea to work with build arguments, so thank you for that. It also uses tags for each PHP version, so we are more inline with the docker naming conventions. Closing this one in favor of #23.

**Comment by laoneo** [comment]
Let me know what you think about #23. And thank you for your work here.

**Comment by astridx** [comment]
I haven't uploaded anything because I wasn't sure yet. I thought about a configuration file that you can use to set which setup you want. For example, I personally don't have an extension that runs under j3 and j4 at the same time. But: I would like to have different PHP and database versions on the web server at the same time.

The way I see it, it doesn't fit your workflow. You don't need different PHP and database versions in the web server, you only do that via tests. For this it is important to you to have j3 and j4 in the web server at the same time. Right?

**Comment by laoneo** [comment]
How would you switch then between PHP versions?
